src/hotspot/share/nmt/memTracker.cpp

```diff
@@ -94,6 +96,8 @@ void MemTracker::initialize() {
     NMTPreInit::print_state(&ls);
     MallocLimitHandler::print_on(&ls);
   }
+
+  NMT_LogRecorder::replay();
```

Comment:
If the pid is saved in the corresponding env-var and there is a recording file, `replay()` will run and play at every time JVM starts (all the JTREG tests for example). Perhaps we don't want this.

Reply:
The user can always unset `NMTBenchmarkRecordedPID` and the code will not activate:

```
void NMT_LogRecorder::replay() {
  char* NMTBenchmarkRecordedPID = getenv("NMTBenchmarkRecordedPID");
  if (NMTBenchmarkRecordedPID != nullptr) {
    int pid = atoi(NMTBenchmarkRecordedPID);
    NMT_MemoryLogRecorder::instance()->replay(pid);
    NMT_VirtualMemoryLogRecorder::instance()->replay(pid);
    os::exit(0);
  }
}

```

I think this is fine, but let me know if it really bothers you so we can give it more thought.